packages/public-api/src/web.ts

```diff
@@ -1,27 +1,109 @@
+// This interface uses by-name style JSON RPC.
 export interface MaskNetworkAPIs {
-    web_echo<T>(arg: T): Promise<T>
+    web_echo<T>(params: { echo: T }): Promise<T>
     /**
      * @returns A fully quantified URL in forms of "holoflows-extension://...." or "moz-extension://...." (based on the platform)
      */
     getDashboardURL(): Promise<string>
 
-    /**
-     * @deprecated Use settings_getLaunchPageSettings()
-     */
-    getSettings(key: 'launchPageSettings'): Promise<LaunchPage>
-
     /**
      * @returns A stringified JSON string.
      * @example [[{ network: "twitter.com", connected: false } ]]
      */
     getConnectedPersonas(): Promise<string>
-    app_isPluginEnabled(pluginID: string): Promise<boolean>
-    app_setPluginStatus(pluginID: string, enabled: boolean): Promise<void>
-    setting_getNetworkTraderProvider(network: NetworkType): Promise<TradeProvider | undefined>
-    setting_setNetworkTraderProvider(network: NetworkType, provider: TradeProvider): Promise<void>
+    app_isPluginEnabled(params: { pluginID: string }): Promise<boolean>
+    app_setPluginStatus(params: { pluginID: string; enabled: boolean }): Promise<void>
+    setting_getNetworkTraderProvider(params: { network: NetworkType }): Promise<TradeProvider | undefined>
+    setting_setNetworkTraderProvider(params: { network: NetworkType; provider: TradeProvider }): Promise<void>
     settings_getTrendingDataSource(): Promise<DataProvider>
-    settings_setTrendingDataSource(provider: DataProvider): Promise<void>
+    settings_setTrendingDataSource(params: { provider: DataProvider }): Promise<void>
     settings_getLaunchPageSettings(): Promise<LaunchPage>
+    settings_getTheme(): Promise<Appearance>
+    settings_setTheme(params: { theme: Appearance }): Promise<void>
+    settings_getLanguage(): Promise<Language>
+    settings_setLanguage(params: { language: Language }): Promise<void>
+    settings_createBackupJson(params: Partial<BackupOptions>): Promise<unknown>
+    settings_getBackupPreviewInfo(params: string): Promise<BackupPreview | undefined>
```

Comment:
I found API inconsistency here. Compare these two:

```ts
    settings_setTheme(params: { theme: Appearance }): Promise<void>
    settings_getBackupPreviewInfo(params: string): Promise<BackupPreview | undefined>
```

Reply:
I will change to object